archive/xar.ksy

```diff
@@ -0,0 +1,124 @@
+meta:
+  id: xar
+  title: XAR (eXtensible ARchive)
+  file-extension:
+    - pkg
+    - xar
+    - xip
+  xref:
+    justsolve: Xar_(Extensible_Archive)
+    mime: application/x-xar
+    pronom: fmt/600
+    wikidata: Q1093556
+  license: CC0-1.0
+  ks-version: 0.9
+  encoding: UTF-8
+  endian: be
+doc: |
+  From Wikipedia:
+
+  "XAR (short for eXtensible ARchive format) is an open source file archiver
+  and the archiver’s file format. It was created within the OpenDarwin project
+  and is used in macOS X 10.5 and up for software installation routines, as
+  well as browser extensions in Safari 5.0 and up."
+
+  It should be noted that there is a different version from Apple which uses
+  the same version number as the (unmaintained) version described here and
+  supports additional checksums, but doesn't have `checksum_alg_name`.
```

Comment:
@armijnhemel Is there a source of that information, a sample file, description or a source code snippet supporting this?

Reply:
Of course. The original:

https://github.com/mackyle/xar/blob/master/xar/include/xar.h.in#L54

and the Apple version:

https://github.com/apple-opensource-mirror/xar/blob/master/xar/include/xar.h.in#L53

The important difference is `struct xar_header_ex`. The header file says:

```
Only used when cksum_alg == XAR_CKSUM_OTHER
```
and includes an extra field `toc_cksum_name[36]`.

It is fairly easy to generate a new test file that shows this, for example:

```
$ xar -c --file-cksum=sha224 -f ls.xar ls
```

Reply:
OK, I took the sentence verbatim so I thought that there is a yet another format that _"supports additional checksums **but** doesn't have `checksum_alg_name`"_. 

I would remove this sentence entirely, I guess this is specified enough by the .ksy logic.